contracts/token/PoolFDT.sol

```diff
@@ -36,11 +64,26 @@ abstract contract PoolFDT is ExtendedFDT {
     function _updateLossesBalance() internal override returns (int256) {
         uint256 _prevLossesTokenBalance = lossesBalance;
 
-        lossesBalance = poolLosses;
+        lossesBalance = lossesSum;
 
         return int256(lossesBalance).sub(int256(_prevLossesTokenBalance));
     }
 
+    /**
+        @dev Withdraws all available funds for a token holder.
+    */
+    function withdrawFunds() public virtual override {
+        uint256 withdrawableFunds = _prepareWithdraw();
+
+        if (withdrawableFunds == uint256(0)) return;
+
+        fundsToken.safeTransfer(msg.sender, withdrawableFunds);
+
+        interestSum = interestSum.sub(withdrawableFunds);
```

Comment:
Take straight from `Pool`, except with a `safeTransfer` instead of a `LiquidityLocker.transfer`.